overlord/snapstate/autorefresh_gating.go

```diff
@@ -162,17 +159,38 @@ func affectedByRefresh(st *state.State, updates []*snap.Info) (map[string]*affec
 				return nil, err
 			}
 			for _, cref := range conns {
-				// affected only if it wasn't optimized out above
 				if all[cref.SlotRef.Snap] != nil {
 					addAffected(cref.SlotRef.Snap, up.InstanceName(), true, false)
 				}
 			}
 		}
+		if up.SnapType == snap.TypeSnapd || up.SnapType == snap.TypeOS {
```

Comment:
you might want to switch the order around and do the plugs&mount case last (they are probably going to be a nop when we add metadata).  Also this one probably merits a comment on the fact that we are only doing this for snapd and core because they are ignored in the loop at 133, otoh any app snap slot will be considered there (irrespective whether it's using mount or not). Another approach would be to merge the loops instead again. But consider the kind of interface when up is snapd or core.

Reply:
Yes, I was considering joining the loop but I felt it would complicate it a bit (and would require these extra checks inside the loop rather than outside). I'd leave it as with the clarifying comment (added).